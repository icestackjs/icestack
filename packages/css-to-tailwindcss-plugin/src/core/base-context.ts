import type { Node } from 'postcss'
import { createGenerator } from './generator'
import { TailwindcssPluginOptions } from '@/types'
import { LayerEnumType } from '@/constants'
// import { TailwindcssPluginOptions } from '@/types'

export class BaseContext {
  layersMap: Map<LayerEnumType, Node[]>
  options: Required<TailwindcssPluginOptions> | undefined
  generator: ReturnType<typeof createGenerator>
  constructor() {
    this.layersMap = new Map<LayerEnumType, Node[]>()

    this.resetLayersMap()

    this.generator = createGenerator()
  }

  getNodes(key: LayerEnumType) {
    const arrRef = this.layersMap.get(key)
    if (Array.isArray(arrRef)) {
      return arrRef
    }
    const arr: Node[] = []
    this.layersMap.set(key, arr)
    return arr
  }

  append(key: LayerEnumType, nodes: Node | Node[]) {
    const arr = this.getNodes(key)
    if (Array.isArray(nodes)) {
      arr.push(...nodes)
    } else {
      arr.push(nodes)
    }
  }

  resetLayersMap() {
    this.layersMap.set('base', [])
    this.layersMap.set('components', [])
    this.layersMap.set('utilities', [])
  }
}
