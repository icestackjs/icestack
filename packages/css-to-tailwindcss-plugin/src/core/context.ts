// dispose base components utilities
import type { Node } from 'postcss'

export type NodeMapKeys = 'base' | 'components' | 'utilities'

export function createContext() {
  const nodeMap = new Map<NodeMapKeys, Node[]>()
  nodeMap.set('base', [])
  nodeMap.set('components', [])
  nodeMap.set('utilities', [])

  function getNodes(key: NodeMapKeys) {
    return nodeMap.get(key)
  }

  function append(key: NodeMapKeys, node: Node) {
    return getNodes(key)?.push(node)
  }

  return {
    append,
    getNodes
  }
}

export type IContext = ReturnType<typeof createContext>
