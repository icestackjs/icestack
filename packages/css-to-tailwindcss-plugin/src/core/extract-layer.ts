import type { PluginCreator } from 'postcss'

const layerNodes = {
  base: null,
  components: null,
  utilities: null
  // variants: null
}

const markedLayerKey = '__tw_layer_name__'

const layerNodesKeys = Object.keys(layerNodes)

export const markLayerPlugin: PluginCreator<object> = () => {
  return {
    postcssPlugin: 'postcss-css-to-tailwindcss-plugin-mark-layer-plugin',
    Once(root) {
      root.walkAtRules('layer', (rule) => {
        if (layerNodesKeys.includes(rule.params)) {
          const layerName = rule.params
          for (const node of rule.nodes) {
            // @ts-ignore
            node[markedLayerKey] = layerName
          }
          rule.parent?.insertBefore(rule, rule.nodes)

          rule.remove()
        }
      })
    }
  }
}

markLayerPlugin.postcss = true

export const extractLayerPlugin: PluginCreator<object> = () => {
  return {
    postcssPlugin: 'postcss-css-to-tailwindcss-plugin-extract-layer-plugin',
    OnceExit(root) {
      // console.log(root)
    }
  }
}

extractLayerPlugin.postcss = true
