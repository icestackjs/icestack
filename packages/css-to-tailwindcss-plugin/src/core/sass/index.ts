import type { Options, StringOptions } from 'sass'

export function tryLoadSass() {
  return import('sass')
}
export function tryLoadSassSync() {
  return require('sass')
}

export function isExtSassFile(extname: string) {
  return extname === '.scss' || extname === '.sass'
}

export async function sassCompile(path: string, sassOptions?: Options<'sync'>) {
  const { compile } = await tryLoadSass()
  const { css } = compile(path, sassOptions)
  return css
}

export function sassCompileSync(path: string, sassOptions?: Options<'sync'>) {
  const { compile } = tryLoadSassSync()
  const { css } = compile(path, sassOptions)
  return css
}

export async function sassCompileString(code: string, sassOptions?: StringOptions<'sync'>) {
  const { compileString } = await tryLoadSass()
  const { css } = compileString(code, sassOptions)
  return css
}
