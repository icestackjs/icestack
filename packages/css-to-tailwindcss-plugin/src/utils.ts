import path from 'node:path'
import fs from 'node:fs'
export function resolvePath(p: string, cwd: string = process.cwd()) {
  return path.isAbsolute(p) ? p : path.resolve(cwd, p)
}

export function ensureDir(dir: string) {
  if (fs.existsSync(dir)) {
    const stat = fs.statSync(dir)
    if (!stat.isDirectory()) {
      fs.mkdirSync(dir)
    }
  } else {
    fs.mkdirSync(dir)
  }
}
