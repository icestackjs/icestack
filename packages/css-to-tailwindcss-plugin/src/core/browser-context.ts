import type { AcceptedPlugin } from 'postcss'
import postcss from 'postcss'
import defu from 'defu'
import { extractLayerPlugin, markLayerPlugin, atRulesRenamePlugin } from './extract-layer'
// import { sassCompileString } from './sass'
import { BaseContext } from './base-context'
import type { IProcessOptions, TailwindcssPluginOptions } from '@/types'

export function getDefaults(): IProcessOptions & { cacheDir?: string } {
  return {
    atImportOptions: {},
    tailwindcssResolved: false,
    withOptions: true,
    withOptionsWalkCSSRuleObject(obj) {
      return obj
    }
  }
}

export function getClientOptions(opts?: IProcessOptions) {
  return defu(opts, getDefaults()) as Required<TailwindcssPluginOptions>
}

export function createContext(opts?: IProcessOptions) {
  const ctx = new BaseContext()
  const options = getClientOptions(opts)
  ctx.options = options
  return {
    ctx,
    getPlugins() {
      // const { tailwindcssConfig, tailwindcssResolved } = options
      const opt = { ctx }
      const plugins: AcceptedPlugin[] = [atRulesRenamePlugin(opt)]
      // if (tailwindcssResolved && tailwindcssConfig) {
      //   const { default: tailwindcss } = await import('tailwindcss')
      //   plugins.push(tailwindcss(tailwindcssConfig))
      // }
      plugins.push(markLayerPlugin(opt))
      plugins.push(extractLayerPlugin(opt))
      options.postcssPlugins?.(plugins)
      return plugins
    },

    async processString(rawCss: string) {
      // const { sassOptions } = options
      const plugins = await this.getPlugins()
      // const css = await sassCompileString(rawCss, sassOptions)
      // for more langs support
      const res = await postcss(plugins).process(rawCss, {
        from: undefined
      })
      return res
    },
    generate: function () {
      const { generatorOptions } = options
      return ctx.generator.generate(ctx, generatorOptions)
    }
  }
}

export type IContext = ReturnType<typeof createContext>
