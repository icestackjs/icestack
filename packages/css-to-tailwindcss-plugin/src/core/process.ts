import fs from 'node:fs/promises'
import type { Config } from 'tailwindcss'
import type { AcceptedPlugin } from 'postcss'
import postcss from 'postcss'
import { markLayerPlugin, extractLayerPlugin } from './extract-layer'

export interface IProcessOptions {
  tailwindcssConfig?:
    | string
    | Config
    | {
        config: string | Config
      }
    | undefined
}

export async function getPlugins(options: IProcessOptions = {}) {
  const { tailwindcssConfig } = options

  const plugins: AcceptedPlugin[] = [markLayerPlugin()]
  if (tailwindcssConfig) {
    const { default: tailwindcss } = await import('tailwindcss')
    plugins.push(tailwindcss(tailwindcssConfig))
  }
  plugins.push(extractLayerPlugin())
  return plugins
}

export async function getCss(entry: string, options: IProcessOptions = {}) {
  const plugins = await getPlugins(options)
  const css = await fs.readFile(entry, 'utf8')
  // @ts-ignore
  const res = await postcss(plugins).process(css)
  return res.css
}
