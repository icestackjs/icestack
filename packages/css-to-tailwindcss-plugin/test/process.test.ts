import { compile } from '@icestack/css2js'
import { fixturesResolve } from './utils'
import { IProcessOptions, getCss, getPlugins } from '@/core/process'
async function process(entry: string, options: IProcessOptions = {}) {
  const plugins = await getPlugins(options)

  const res = await compile({
    path: entry,
    plugins
  })
  return res
}

describe('process', () => {
  it('default.css', async () => {
    const obj = await process(fixturesResolve('default.css'))
    // const obj2 = await process(fixturesResolve('default.scss'))
    // expect(obj).toEqual(obj2)
    expect(obj).toMatchSnapshot()
  })
  it('default.scss', async () => {
    const obj = await process(fixturesResolve('default.scss'))
    // const obj2 = await process(fixturesResolve('default.scss'))
    // expect(obj).toEqual(obj2)
    expect(obj).toMatchSnapshot()
  })

  it('common.css', async () => {
    const p = fixturesResolve('common.css')
    const obj = await process(p, {
      tailwindcssConfig: fixturesResolve('config/tailwind.config.js')
    })
    expect(obj).toMatchSnapshot()
    const css = await getCss(p, {
      tailwindcssConfig: fixturesResolve('config/tailwind.config.js')
    })
    expect(css).toMatchSnapshot()
  })

  it('common.css only extract layer', async () => {
    const p = fixturesResolve('common.css')
    const obj = await process(p, {
      // tailwindcssConfig: fixturesResolve('config/tailwind.config.js')
    })

    expect(obj).toMatchSnapshot()
    const css = await getCss(p)
    expect(css).toMatchSnapshot()
  })

  // it('defaults are same', async () => {
  //   const obj0 = await process(fixturesResolve('default.css'))
  //   const obj1 = await process(fixturesResolve('default.scss'))
  //   expect(obj0).toEqual(obj1)
  //   // const obj2 = await process(fixturesResolve('default.scss'))
  //   //

  // })
})
