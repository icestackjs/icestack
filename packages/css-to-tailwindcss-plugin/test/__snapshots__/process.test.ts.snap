// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`process > common.css 1`] = `
{
  "*, ::before, ::after": {
    "--tw-backdrop-blur": " ",
    "--tw-backdrop-brightness": " ",
    "--tw-backdrop-contrast": " ",
    "--tw-backdrop-grayscale": " ",
    "--tw-backdrop-hue-rotate": " ",
    "--tw-backdrop-invert": " ",
    "--tw-backdrop-opacity": " ",
    "--tw-backdrop-saturate": " ",
    "--tw-backdrop-sepia": " ",
    "--tw-blur": " ",
    "--tw-border-spacing-x": "0",
    "--tw-border-spacing-y": "0",
    "--tw-brightness": " ",
    "--tw-contrast": " ",
    "--tw-drop-shadow": " ",
    "--tw-gradient-from-position": " ",
    "--tw-gradient-to-position": " ",
    "--tw-gradient-via-position": " ",
    "--tw-grayscale": " ",
    "--tw-hue-rotate": " ",
    "--tw-invert": " ",
    "--tw-numeric-figure": " ",
    "--tw-numeric-fraction": " ",
    "--tw-numeric-spacing": " ",
    "--tw-ordinal": " ",
    "--tw-pan-x": " ",
    "--tw-pan-y": " ",
    "--tw-pinch-zoom": " ",
    "--tw-ring-color": "rgb(59 130 246 / 0.5)",
    "--tw-ring-inset": " ",
    "--tw-ring-offset-color": "#fff",
    "--tw-ring-offset-shadow": "0 0 #0000",
    "--tw-ring-offset-width": "0px",
    "--tw-ring-shadow": "0 0 #0000",
    "--tw-rotate": "0",
    "--tw-saturate": " ",
    "--tw-scale-x": "1",
    "--tw-scale-y": "1",
    "--tw-scroll-snap-strictness": "proximity",
    "--tw-sepia": " ",
    "--tw-shadow": "0 0 #0000",
    "--tw-shadow-colored": "0 0 #0000",
    "--tw-skew-x": "0",
    "--tw-skew-y": "0",
    "--tw-slashed-zero": " ",
    "--tw-translate-x": "0",
    "--tw-translate-y": "0",
  },
  ".card": {
    "backgroundColor": "#fff",
    "borderRadius": "0.5rem",
    "boxShadow": "0 20px 25px -5px rgb(0 0 0 / 0.1), 0 8px 10px -6px rgb(0 0 0 / 0.1)",
    "padding": "1.5rem",
  },
  ".content-auto": {
    "contentVisibility": "\\"auto\\"",
  },
  "::backdrop": {
    "--tw-backdrop-blur": " ",
    "--tw-backdrop-brightness": " ",
    "--tw-backdrop-contrast": " ",
    "--tw-backdrop-grayscale": " ",
    "--tw-backdrop-hue-rotate": " ",
    "--tw-backdrop-invert": " ",
    "--tw-backdrop-opacity": " ",
    "--tw-backdrop-saturate": " ",
    "--tw-backdrop-sepia": " ",
    "--tw-blur": " ",
    "--tw-border-spacing-x": "0",
    "--tw-border-spacing-y": "0",
    "--tw-brightness": " ",
    "--tw-contrast": " ",
    "--tw-drop-shadow": " ",
    "--tw-gradient-from-position": " ",
    "--tw-gradient-to-position": " ",
    "--tw-gradient-via-position": " ",
    "--tw-grayscale": " ",
    "--tw-hue-rotate": " ",
    "--tw-invert": " ",
    "--tw-numeric-figure": " ",
    "--tw-numeric-fraction": " ",
    "--tw-numeric-spacing": " ",
    "--tw-ordinal": " ",
    "--tw-pan-x": " ",
    "--tw-pan-y": " ",
    "--tw-pinch-zoom": " ",
    "--tw-ring-color": "rgb(59 130 246 / 0.5)",
    "--tw-ring-inset": " ",
    "--tw-ring-offset-color": "#fff",
    "--tw-ring-offset-shadow": "0 0 #0000",
    "--tw-ring-offset-width": "0px",
    "--tw-ring-shadow": "0 0 #0000",
    "--tw-rotate": "0",
    "--tw-saturate": " ",
    "--tw-scale-x": "1",
    "--tw-scale-y": "1",
    "--tw-scroll-snap-strictness": "proximity",
    "--tw-sepia": " ",
    "--tw-shadow": "0 0 #0000",
    "--tw-shadow-colored": "0 0 #0000",
    "--tw-skew-x": "0",
    "--tw-skew-y": "0",
    "--tw-slashed-zero": " ",
    "--tw-translate-x": "0",
    "--tw-translate-y": "0",
  },
  "h1": {
    "fontSize": "1.5rem",
  },
  "h2": {
    "fontSize": "1.25rem",
  },
}
`;

exports[`process > common.css 2`] = `
"*, ::before, ::after {
    --tw-border-spacing-x: 0;
    --tw-border-spacing-y: 0;
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-pan-x:  ;
    --tw-pan-y:  ;
    --tw-pinch-zoom:  ;
    --tw-scroll-snap-strictness: proximity;
    --tw-gradient-from-position:  ;
    --tw-gradient-via-position:  ;
    --tw-gradient-to-position:  ;
    --tw-ordinal:  ;
    --tw-slashed-zero:  ;
    --tw-numeric-figure:  ;
    --tw-numeric-spacing:  ;
    --tw-numeric-fraction:  ;
    --tw-ring-inset:  ;
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgb(59 130 246 / 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
    --tw-shadow-colored: 0 0 #0000;
    --tw-blur:  ;
    --tw-brightness:  ;
    --tw-contrast:  ;
    --tw-grayscale:  ;
    --tw-hue-rotate:  ;
    --tw-invert:  ;
    --tw-saturate:  ;
    --tw-sepia:  ;
    --tw-drop-shadow:  ;
    --tw-backdrop-blur:  ;
    --tw-backdrop-brightness:  ;
    --tw-backdrop-contrast:  ;
    --tw-backdrop-grayscale:  ;
    --tw-backdrop-hue-rotate:  ;
    --tw-backdrop-invert:  ;
    --tw-backdrop-opacity:  ;
    --tw-backdrop-saturate:  ;
    --tw-backdrop-sepia:  ;
}

::backdrop {
    --tw-border-spacing-x: 0;
    --tw-border-spacing-y: 0;
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    --tw-pan-x:  ;
    --tw-pan-y:  ;
    --tw-pinch-zoom:  ;
    --tw-scroll-snap-strictness: proximity;
    --tw-gradient-from-position:  ;
    --tw-gradient-via-position:  ;
    --tw-gradient-to-position:  ;
    --tw-ordinal:  ;
    --tw-slashed-zero:  ;
    --tw-numeric-figure:  ;
    --tw-numeric-spacing:  ;
    --tw-numeric-fraction:  ;
    --tw-ring-inset:  ;
    --tw-ring-offset-width: 0px;
    --tw-ring-offset-color: #fff;
    --tw-ring-color: rgb(59 130 246 / 0.5);
    --tw-ring-offset-shadow: 0 0 #0000;
    --tw-ring-shadow: 0 0 #0000;
    --tw-shadow: 0 0 #0000;
    --tw-shadow-colored: 0 0 #0000;
    --tw-blur:  ;
    --tw-brightness:  ;
    --tw-contrast:  ;
    --tw-grayscale:  ;
    --tw-hue-rotate:  ;
    --tw-invert:  ;
    --tw-saturate:  ;
    --tw-sepia:  ;
    --tw-drop-shadow:  ;
    --tw-backdrop-blur:  ;
    --tw-backdrop-brightness:  ;
    --tw-backdrop-contrast:  ;
    --tw-backdrop-grayscale:  ;
    --tw-backdrop-hue-rotate:  ;
    --tw-backdrop-invert:  ;
    --tw-backdrop-opacity:  ;
    --tw-backdrop-saturate:  ;
    --tw-backdrop-sepia:  ;
}

h1 {
    font-size: 1.5rem;
  }

h2 {
    font-size: 1.25rem;
  }

.card {
    background-color: #fff;
    border-radius: 0.5rem;
    padding: 1.5rem;
    box-shadow: 0 20px 25px -5px rgb(0 0 0 / 0.1), 0 8px 10px -6px rgb(0 0 0 / 0.1);
  }

.content-auto {
    content-visibility: \\"auto\\";
  }
"
`;

exports[`process > common.css only extract layer 1`] = `
{
  ".card": {
    "backgroundColor": "theme(\\"colors.white\\")",
    "borderRadius": "theme(\\"borderRadius.lg\\")",
    "boxShadow": "theme(\\"boxShadow.xl\\")",
    "padding": "theme(\\"spacing.6\\")",
  },
  ".content-auto": {
    "contentVisibility": "\\"auto\\"",
  },
  "@tailwind base": true,
  "@tailwind components": true,
  "@tailwind utilities": true,
  "h1": {
    "fontSize": "theme(\\"fontSize.2xl\\")",
  },
  "h2": {
    "fontSize": "theme(\\"fontSize.xl\\")",
  },
}
`;

exports[`process > common.css only extract layer 2`] = `
"@tailwind base;
@tailwind components;
@tailwind utilities;

h1 {
    font-size: theme(\\"fontSize.2xl\\");
  }

h2 {
    font-size: theme(\\"fontSize.xl\\");
  }

.card {
    background-color: theme(\\"colors.white\\");
    border-radius: theme(\\"borderRadius.lg\\");
    padding: theme(\\"spacing.6\\");
    box-shadow: theme(\\"boxShadow.xl\\");
  }

.content-auto {
    content-visibility: \\"auto\\";
  }
"
`;

exports[`process > default.css 1`] = `
{
  "*,
::before,
::after": {
    "borderColor": "theme(\\"borderColor.DEFAULT\\", currentColor)",
    "borderStyle": "solid",
    "borderWidth": "0",
    "boxSizing": "border-box",
  },
  ":-moz-focusring": {
    "outline": "auto",
  },
  ":-moz-ui-invalid": {
    "boxShadow": "none",
  },
  "::-webkit-file-upload-button": {
    "WebkitAppearance": "button",
    "font": "inherit",
  },
  "::-webkit-inner-spin-button,
::-webkit-outer-spin-button": {
    "height": "auto",
  },
  "::-webkit-search-decoration": {
    "WebkitAppearance": "none",
  },
  "::before,
::after": {
    "--tw-content": "\\"\\"",
  },
  ":disabled": {
    "cursor": "default",
  },
  "[hidden]": {
    "display": "none",
  },
  "[type=\\"search\\"]": {
    "WebkitAppearance": "textfield",
    "outlineOffset": "-2px",
  },
  "a": {
    "color": "inherit",
    "textDecoration": "inherit",
  },
  "abbr:where([title])": {
    "textDecoration": "underline dotted",
  },
  "b,
strong": {
    "fontWeight": "bolder",
  },
  "blockquote,
dl,
dd,
h1,
h2,
h3,
h4,
h5,
h6,
hr,
figure,
p,
pre": {
    "margin": "0",
  },
  "body": {
    "lineHeight": "inherit",
    "margin": "0",
  },
  "button,
[role=\\"button\\"]": {
    "cursor": "pointer",
  },
  "button,
[type=\\"button\\"],
[type=\\"reset\\"],
[type=\\"submit\\"]": {
    "WebkitAppearance": "button",
    "backgroundColor": "transparent",
    "backgroundImage": "none",
  },
  "button,
input,
optgroup,
select,
textarea": {
    "color": "inherit",
    "fontFamily": "inherit",
    "fontFeatureSettings": "inherit",
    "fontSize": "100%",
    "fontVariationSettings": "inherit",
    "fontWeight": "inherit",
    "lineHeight": "inherit",
    "margin": "0",
    "padding": "0",
  },
  "button,
select": {
    "textTransform": "none",
  },
  "code,
kbd,
samp,
pre": {
    "fontFamily": "theme(
    \\"fontFamily.mono\\",
    ui-monospace,
    SFMono-Regular,
    Menlo,
    Monaco,
    Consolas,
    \\"Liberation Mono\\",
    \\"Courier New\\",
    monospace
  )",
    "fontSize": "1em",
  },
  "dialog": {
    "padding": "0",
  },
  "fieldset": {
    "margin": "0",
    "padding": "0",
  },
  "h1,
h2,
h3,
h4,
h5,
h6": {
    "fontSize": "inherit",
    "fontWeight": "inherit",
  },
  "hr": {
    "borderTopWidth": "1px",
    "color": "inherit",
    "height": "0",
  },
  "html": {
    "MozTabSize": "4",
    "WebkitTextSizeAdjust": "100%",
    "fontFamily": "theme(
    \\"fontFamily.sans\\",
    ui-sans-serif,
    system-ui,
    -apple-system,
    BlinkMacSystemFont,
    \\"Segoe UI\\",
    Roboto,
    \\"Helvetica Neue\\",
    Arial,
    \\"Noto Sans\\",
    sans-serif,
    \\"Apple Color Emoji\\",
    \\"Segoe UI Emoji\\",
    \\"Segoe UI Symbol\\",
    \\"Noto Color Emoji\\"
  )",
    "fontFeatureSettings": "theme(\\"fontFamily.sans[1].fontFeatureSettings\\", normal)",
    "fontVariationSettings": "theme(\\"fontFamily.sans[1].fontVariationSettings\\", normal)",
    "lineHeight": 1.5,
    "tabSize": 4,
  },
  "img,
svg,
video,
canvas,
audio,
iframe,
embed,
object": {
    "display": "block",
    "verticalAlign": "middle",
  },
  "img,
video": {
    "height": "auto",
    "maxWidth": "100%",
  },
  "input::placeholder,
textarea::placeholder": {
    "color": "theme(\\"colors.gray.400\\", #9ca3af)",
    "opacity": 1,
  },
  "legend": {
    "padding": "0",
  },
  "ol,
ul,
menu": {
    "listStyle": "none",
    "margin": "0",
    "padding": "0",
  },
  "progress": {
    "verticalAlign": "baseline",
  },
  "small": {
    "fontSize": "80%",
  },
  "sub": {
    "bottom": "-0.25em",
  },
  "sub,
sup": {
    "fontSize": "75%",
    "lineHeight": 0,
    "position": "relative",
    "verticalAlign": "baseline",
  },
  "summary": {
    "display": "list-item",
  },
  "sup": {
    "top": "-0.5em",
  },
  "table": {
    "borderCollapse": "collapse",
    "borderColor": "inherit",
    "textIndent": "0",
  },
  "textarea": {
    "resize": "vertical",
  },
}
`;

exports[`process > default.scss 1`] = `
{
  "*,
::before,
::after": {
    "borderColor": "theme(\\"borderColor.DEFAULT\\", currentColor)",
    "borderStyle": "solid",
    "borderWidth": "0",
    "boxSizing": "border-box",
  },
  ":-moz-focusring": {
    "outline": "auto",
  },
  ":-moz-ui-invalid": {
    "boxShadow": "none",
  },
  "::-webkit-file-upload-button": {
    "WebkitAppearance": "button",
    "font": "inherit",
  },
  "::-webkit-inner-spin-button,
::-webkit-outer-spin-button": {
    "height": "auto",
  },
  "::-webkit-search-decoration": {
    "WebkitAppearance": "none",
  },
  "::before,
::after": {
    "--tw-content": "\\"\\"",
  },
  ":disabled": {
    "cursor": "default",
  },
  "[hidden]": {
    "display": "none",
  },
  "[type=search]": {
    "WebkitAppearance": "textfield",
    "outlineOffset": "-2px",
  },
  "a": {
    "color": "inherit",
    "textDecoration": "inherit",
  },
  "abbr:where([title])": {
    "textDecoration": "underline dotted",
  },
  "b,
strong": {
    "fontWeight": "bolder",
  },
  "blockquote,
dl,
dd,
h1,
h2,
h3,
h4,
h5,
h6,
hr,
figure,
p,
pre": {
    "margin": "0",
  },
  "body": {
    "lineHeight": "inherit",
    "margin": "0",
  },
  "button,
[role=button]": {
    "cursor": "pointer",
  },
  "button,
[type=button],
[type=reset],
[type=submit]": {
    "WebkitAppearance": "button",
    "backgroundColor": "transparent",
    "backgroundImage": "none",
  },
  "button,
input,
optgroup,
select,
textarea": {
    "color": "inherit",
    "fontFamily": "inherit",
    "fontFeatureSettings": "inherit",
    "fontSize": "100%",
    "fontVariationSettings": "inherit",
    "fontWeight": "inherit",
    "lineHeight": "inherit",
    "margin": "0",
    "padding": "0",
  },
  "button,
select": {
    "textTransform": "none",
  },
  "code,
kbd,
samp,
pre": {
    "fontFamily": "theme(\\"fontFamily.mono\\", ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, \\"Liberation Mono\\", \\"Courier New\\", monospace)",
    "fontSize": "1em",
  },
  "dialog": {
    "padding": "0",
  },
  "fieldset": {
    "margin": "0",
    "padding": "0",
  },
  "h1,
h2,
h3,
h4,
h5,
h6": {
    "fontSize": "inherit",
    "fontWeight": "inherit",
  },
  "hr": {
    "borderTopWidth": "1px",
    "color": "inherit",
    "height": "0",
  },
  "html": {
    "MozTabSize": "4",
    "WebkitTextSizeAdjust": "100%",
    "fontFamily": "theme(\\"fontFamily.sans\\", ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\")",
    "fontFeatureSettings": "theme(\\"fontFamily.sans[1].fontFeatureSettings\\", normal)",
    "fontVariationSettings": "theme(\\"fontFamily.sans[1].fontVariationSettings\\", normal)",
    "lineHeight": 1.5,
    "tabSize": 4,
  },
  "img,
svg,
video,
canvas,
audio,
iframe,
embed,
object": {
    "display": "block",
    "verticalAlign": "middle",
  },
  "img,
video": {
    "height": "auto",
    "maxWidth": "100%",
  },
  "input::placeholder,
textarea::placeholder": {
    "color": "theme(\\"colors.gray.400\\", #9ca3af)",
    "opacity": 1,
  },
  "legend": {
    "padding": "0",
  },
  "ol,
ul,
menu": {
    "listStyle": "none",
    "margin": "0",
    "padding": "0",
  },
  "progress": {
    "verticalAlign": "baseline",
  },
  "small": {
    "fontSize": "80%",
  },
  "sub": {
    "bottom": "-0.25em",
  },
  "sub,
sup": {
    "fontSize": "75%",
    "lineHeight": 0,
    "position": "relative",
    "verticalAlign": "baseline",
  },
  "summary": {
    "display": "list-item",
  },
  "sup": {
    "top": "-0.5em",
  },
  "table": {
    "borderCollapse": "collapse",
    "borderColor": "inherit",
    "textIndent": "0",
  },
  "textarea": {
    "resize": "vertical",
  },
}
`;
